types/react-beautiful-dnd/index.d.ts

```diff
@@ -34,73 +174,124 @@ export interface DropResult {
-    placeholder?: React.ReactElement<any>;
+    // When we do `position: fixed` the element looses its normal dimensions,
+    // especially if using flexbox. We set the width and height manually to
+    // ensure the element has the same dimensions as before it started dragging
+    width: number;
+    height: number;
+
+    // The width and height values take into account whether the original element
+    // used `box-sizing: content-box` or `box-sizing: border-box`
+    // Because we are setting the width and height directly we want to ensure that
+    // these are the actual values applied
+    boxSizing: 'border-box';
+
+    // We initially position the element in the same *visual spot* as when it started.
+    // This means that these values *exclude* the original margins so that element remains
+    // in the same visual position - even though now it has no margins
+    top: number;
+    left: number;
+
+    // We clear any top or left margins on the element to ensure it does not push
+    // the element positioned with the top/left position (which is margin aware).
+    // We also clear the margin right / bottom. This has no positioning impact,
+    // but it is cleanest to just remove all the margins rather than only the top and left.
+    margin: 0;
+
+    // We need to opt out of the shared global style that is being applied to
+    // all draggables. The movement of moving draggables is either not animated
+    // or handled by react-motion.
+    transition: 'none';
+
+    // Move the element in response to a user dragging
+    transform?: Maybe<string>;
+
+    // When dragging or dropping we control the z-index to ensure that
+    // the layering is correct
+    zIndex: ZIndex;
 }
 
-export interface DroppableStateSnapshot {
-    isDraggingOver: boolean;
+export interface NotDraggingStyle {
+    transform?: Maybe<string>;
+    // null: use the global animation style
+    // none: skip animation (used in certain displacement situations)
+    transition: null | 'none';
 }
 
-export interface DroppableProps {
-    droppableId: DroppableId;
-    type?: TypeId;
-    isDropDisabled?: boolean;
-    direction?: 'vertical' | 'horizontal';
-    children(provided: DroppableProvided, snapshot: DroppableStateSnapshot): React.ReactElement<any>;
+export type DraggableStyle = DraggingStyle | NotDraggingStyle;
+
+// Props that can be spread onto the element directly
+export interface DraggableProps {
+    // inline style
+    style?: Maybe<DraggableStyle>;
+    // used for shared global styles
+    'data-react-beautiful-dnd-draggable': string;
 }
 
-export class Droppable extends React.Component<DroppableProps> {}
+export interface DraggableProvided {
+    draggableProps: DraggableProps;
+    // will be null if the draggable is disabled
+    dragHandleProps?: Maybe<DragHandleProps>;
+    // The following props will be removed once we move to react 16
+    innerRef(element?: Maybe<HTMLElement>): void;
+    placeholder?: Maybe<Node>;
```

Comment:
This should be `React.ReactNode`, not `Node`.

Reply:
Thanks for the catch! Will fix